src/main/java/io/cdap/delta/bigquery/BigQueryAssessor.java

```diff
@@ -53,10 +58,41 @@ public TableAssessment assess(StandardizedTableDetail tableDetail) {
       }
     }
     List<Problem> problems = new ArrayList<>();
+    String dbName = tableDetail.getDatabase();
+    String tableName = tableDetail.getTable();
     if (tableDetail.getPrimaryKey().isEmpty()) {
-      problems.add(new Problem("Missing Primary Key", "Tables must have a primary key in order to be replicated.",
-                               "Please alter the table to use a primary key, or select a different table", ""));
+      problems.add(
+        new Problem("Missing Primary Key",
+                    String.format("Table '%s' in database '%s' must have a primary key in order to be replicated",
+                                  dbName, tableName),
+                    "Please alter the table to use a primary key, or select a different table",
+                    "Not able to replicate this table in BigQuery side"));
+    }
+
+    String normalizedDBName = BigQueryEventConsumer.normalize(dbName);
+    String normalizedTableName = BigQueryEventConsumer.normalize(tableName);
+    String normalizedStagingTableName = BigQueryEventConsumer.normalize(stagingTablePrefix + tableName);
+    if (dbName.length() > BigQueryEventConsumer.MAX_LENGTH || !dbName.matches(BigQueryEventConsumer.VALID_NAME_REGEX)) {
```

Comment:
here it's simpler and safer (in case normalize() logic changes) to just check if the dbName is not equal to the normalizedDBName

Reply:
=> "in case normalize() logic changes"
True, this is a very good suggestion, will do that.